plugins/statsd/statsd.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2017 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	plugin "github.com/dcos/dcos-metrics/plugins"
+	"github.com/dcos/dcos-metrics/producers"
+	"github.com/etsy/statsd/examples/go"
+	"github.com/urfave/cli"
+)
+
+var (
+	pluginFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:   "statsd-udp-host",
+			Usage:  "The address of the statsd server",
+			EnvVar: "STATSD_UDP_HOST",
+			Value:  "127.0.0.1",
+		},
+		cli.IntFlag{
+			Name:   "statsd-udp-port",
+			Usage:  "The statsdPort of the statsd server",
+			EnvVar: "STATSD_UDP_PORT",
+			Value:  8125,
+		},
+	}
+	statsdClient *statsd.StatsdClient
+)
+
+func main() {
+	log.Info("Starting statsd DC/OS metrics plugin")
+
+	statsdPlugin, err := plugin.New(
+		plugin.Name("statsd"),
+		plugin.ExtraFlags(pluginFlags),
+		plugin.ConnectorFunc(statsdConnector))
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Fatal(statsdPlugin.StartPlugin())
+}
+
+// statsdConnector is the method called by the plugin every time it retrieves
+// metrics from the API.
+func statsdConnector(metrics []producers.MetricsMessage, c *cli.Context) error {
+	if statsdClient == nil {
```

Comment:
I'd probably wrap this initialization in a `sync.Once` since you're mutating global state with a function that something else is calling.